components/layout_2020/positioned.rs

```diff
@@ -521,27 +530,45 @@ impl HoistedAbsolutelyPositionedBox {
             true => style.clone_align_self().0 .0,
             false => shared_fragment.resolved_alignment.block,
         };
-        let block_axis_solver = AbsoluteAxisSolver {
+        let mut block_axis_solver = AbsoluteAxisSolver {
             axis: AxisDirection::Block,
             containing_size: cbbs,
             padding_border_sum: pbm.padding_border_sums.block,
             computed_margin_start: pbm.margin.block_start,
             computed_margin_end: pbm.margin.block_end,
+            computed_size: computed_size.block,
+            computed_min_size: computed_min_size.block,
+            computed_max_size: computed_max_size.block,
             avoid_negative_margin_start: false,
             box_offsets: block_box_offsets,
             static_position_rect_axis: static_position_rect.get_axis(AxisDirection::Block),
             alignment: block_alignment,
             flip_anchor: false,
         };
 
-        let mut inline_axis = inline_axis_solver.solve_for_size(computed_size.inline);
-        let mut block_axis = block_axis_solver.solve_for_size(computed_size.block);
+        // We can solve the inline axis, but the block one can depend on layout results,
+        // so we may have to resolve it properly later on.
+        let mut inline_axis = inline_axis_solver.solve(Some(|| {
+            let block_size = computed_size.block.map(|v| {
+                v.clamp_between_extremums(computed_min_size.block, computed_max_size.block)
+            });
+            let containing_block_for_children =
+                IndefiniteContainingBlock::new_for_style_and_block_size(&style, block_size);
+            context
+                .inline_content_sizes(
+                    layout_context,
+                    &containing_block_for_children,
+                    &containing_block.into(),
+                )
+                .sizes
+        }));
+        let mut block_axis = block_axis_solver.solve_tentativelly();
```

Comment:
Nit: Typo in the word "tentatively" along with the obligatory apologies for the English language.